debugging/coreclr/dockerManager.ts

```diff
@@ -283,6 +276,61 @@ export class DefaultDockerManager implements DockerManager {
 
     private static readonly ProgramFilesEnvironmentVariable: string = 'ProgramFiles';
 
+    private static compareProperty<T, U>(obj1: T | undefined, obj2: T | undefined, getter: (obj: T) => (U | undefined)): boolean {
+        const prop1 = obj1 ? getter(obj1) : undefined;
+        const prop2 = obj2 ? getter(obj2) : undefined;
+
+        return prop1 === prop2;
+    }
+
+    private static compareDictionary<T>(obj1: T | undefined, obj2: T | undefined, getter: (obj: T) => ({ [key: string]: string } | undefined)): boolean {
```

Comment:
We should be able to find an existing way in nodejs to compare objects rather than implementing your own. For example [deep-equal](https://www.npmjs.com/package/deep-equal) is probably a good option

Reply:
I looked at `deep-equal`, but it didn't quite have the semantics I was looking for (less absolute equality as "effective" equality in the context of the Docker commands.  For example, when specifying an array of custom tag names for an image, for the purposes of deciding whether to reuse an image for debugging, the order or those tags is irrelevant.

Reply:
Fair enough about array equality, but I think you should still use `deep-equal` to replace `compareDictionary` since dictionaries have no concept of order (right?). Aka I'm fine if you don't call it on the options itself, but you could at least call it on 'args' and 'labels'

Reply:
Happy to switch to `deep-equal` for this purpose; what's the ceremony around pulling in additional 3rd-party dependencies?